coord/src/rmr/rq.c

```diff
@@ -27,21 +28,71 @@ typedef struct MRWorkQueue {
   uv_async_t async;
 } MRWorkQueue;
 
+uv_thread_t loop_th;
+static char loop_th_running = false;
+static char loop_th_ready = false;
+uv_timer_t topologyTimer;
+uv_async_t topologyAsync;
+struct queueItem *pendingTopo = NULL;
+static inline struct queueItem *exchangePendingTopo(struct queueItem *newTopo) {
+  return __atomic_exchange_n(&pendingTopo, newTopo, __ATOMIC_SEQ_CST);
+}
+
+extern RedisModuleCtx *RSDummyContext;
+
+static void topologyTimerCB(uv_timer_t *timer) {
+  uv_async_send(&topologyAsync);
```

Comment:
Why not just check the connection here? Why we need a shared function for check connection and update topology?

Reply:
we can